sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -572,21 +560,14 @@ static void emitBuiltProgramInfo(const pi_program &Prog,
 
 sycl::detail::pi::PiProgram ProgramManager::getBuiltPIProgram(
     const ContextImplPtr &ContextImpl, const DeviceImplPtr &DeviceImpl,
-    const std::string &KernelName, const program_impl *Prg,
-    bool JITCompilationIsRequired) {
+    const std::string &KernelName, bool JITCompilationIsRequired) {
   KernelProgramCache &Cache = ContextImpl->getKernelProgramCache();
 
-  std::string CompileOpts;
-  std::string LinkOpts;
-  if (Prg) {
-    CompileOpts = Prg->get_build_options();
-  }
-
-  applyOptionsFromEnvironment(CompileOpts, LinkOpts);
+  std::string_view CompileOpts, LinkOpts;
 
+  CompileOpts = getCompileOptionsFromEnvironment();
+  LinkOpts = getLinkOptionsFromEnvironment();
```

Comment:
Now that the program_impl parameter is gone, it appears that the same rationale that applied to link options now applies to compile options as well: they are either empty or set by the environment variable (which means they stay the same for the entire program lifecycle) for the purposes of caching. Can we just remove the build options component of the cache key completely then?

Reply:
> Just to be sure- do you mean we should now remove CompileOpts from CacheKey as well like LinkOpts?
Yes, it looks we can do that to me.
> If so, do we still see any value in switching to string_view then?
Assuming that can be done, then no, that alone should take care of the build options related overhead for kernel lookup. We'll still be working with string build options while actually building programs, but we have to do string concatenation there anyway.

Of course, there's still value in switching kernel name cache key component to string views, but that's outside the scope of this PR.

Reply:
Okay so in the recent commit, I removed CompileOpts from the CacheKey and most of the tests pass. The two failures reported here don't seem related to my changes as I can't reproduce them locally. I guess then we don't need these string_view changes anymore and I should probably discard this PR and create a new one with just the CacheKey changes?

Reply:
@sergey-semenov - I created a draft PR where I removed CompileOpts and LinkOpts from the CacheKey here: https://github.com/intel/llvm/pull/11351